### PR TITLE
[r] Initial support for iterated reads from SOMAReader

### PR DESCRIPTION
**Pull request by eddelbuettel**
This PR adds (initial, limited in that the row / column selection is missing) support for reading SOMA objects in chunks.

### CODE REVIEW

**Review by aaronwolen — APPROVED** [review]
Noted one typo in the docs but otherwise looks awesome!

File: apis/r/src/riterator.cpp [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+    }
+}
+
+//' Iterator-Style Access to SOMA Array via SOMAReader
+//'
+//' The `sr_*` functions provide low-level access to an instance of the SOMAReader
+//' class so that iterative access over parts of a (large) array is possible.
+//' \describe{
+//'   \item{\code{sr_setup}}{instantiates and by default also submits a query}
+//'   \item{\code{sr_complete}}{checks is more data is available}
```
[the remark this answers is not in the document]
Comment (eddelbuettel) [reply apis/r/src/riterator.cpp]: Great catch, now fixed.

**Review by gspowley — APPROVED** [review]
LGTM!

**Comment by eddelbuettel** [comment]
@aaron I adjusted the (demo only, not running) example for the same logic issue discussed on slack that hits for smaller 'one bite' data sets.   Should be good to go now.

**Review by johnkerl — APPROVED** [review]

**Pull request merged**